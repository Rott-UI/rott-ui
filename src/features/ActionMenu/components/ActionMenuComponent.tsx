// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
