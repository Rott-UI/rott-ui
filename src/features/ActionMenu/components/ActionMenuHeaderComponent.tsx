// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
