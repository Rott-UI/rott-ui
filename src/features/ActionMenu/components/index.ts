export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
