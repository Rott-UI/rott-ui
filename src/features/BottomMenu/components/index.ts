export * from './BottomMenu'
export * from './BottomMenuItem'
