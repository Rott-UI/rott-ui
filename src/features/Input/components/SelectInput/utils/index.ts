export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
