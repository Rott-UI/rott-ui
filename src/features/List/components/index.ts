export * from './List'
export * from './FavoriteSwipeComponent'
