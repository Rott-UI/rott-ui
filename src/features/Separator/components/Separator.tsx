// React Imports
import React, {FC} from 'react'
import {View, StyleSheet, ViewProps} from 'react-native'

// Component Imports
import {SeparatorStyles} from '../styles'
import {CommonUiProps} from '../../models'

interface SeparatorProps extends ViewProps, CommonUiProps {
  height?: number | string
  width?: number | string
  orientation?: 'vertical' | 'horizontal'
  opacity?: number
}

export const Separator: FC<SeparatorProps> = ({
  height = 1,
  width = 1,
  style,
  size,
  orientation = 'horizontal',
  opacity = 1,
  ...props
}) => (
  <View
    style={StyleSheet.flatten([
      SeparatorStyles({
        height: orientation === 'vertical' && size ? undefined : height,
        width: orientation === 'horizontal' && size ? undefined : width,
        size,
        orientation,
        opacity,
        ...props,
      }).defaultSeparator,
      style,
    ])}
    {...props}
  />
)
