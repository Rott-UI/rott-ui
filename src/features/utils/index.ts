export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
