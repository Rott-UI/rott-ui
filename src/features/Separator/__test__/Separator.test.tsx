// Feature Imports
import {Separator} from '../components'

// Util and Lib Imports
import {act, render} from './../../../utils'
import {commonUiTestExtension} from '../../utils'

const testId = {
  separatorTestId: 'separator-test-id',
}

describe('Separator -> Custom Component', () => {
  it('separator component render olmalı ve snapshot eşleşmeli', () => {
    const rendered = render(<Separator />)

    expect(rendered).toMatchSnapshot()
  })

  it('separator component default style ile render olmalı', async () => {
    const {separatorTestId} = testId
    const {getByTestId} = render(
      <Separator testID={separatorTestId} size='full' orientation='horizontal' />
    )

    await act(() => {
      const separatorElement = getByTestId(separatorTestId)

      expect(separatorElement).toBeOnTheScreen()
      expect(separatorElement).toHaveStyle({
        height: 2, // Default height
        width: '100%', // Default width
        opacity: 1,
      })
    })
  })

  it('separator component horizontal render olmalı', () => {
    const {separatorTestId} = testId
    const {getByTestId} = render(
      <Separator testID={separatorTestId} size='full' orientation='horizontal' height={'100%'} />
    )

    const separatorElement = getByTestId(separatorTestId)

    expect(separatorElement).toHaveStyle({width: '100%'})
  })

  it('separator component vertical render olmalı', () => {
    const {getByTestId} = render(<Separator orientation='vertical' testID='separator-test-id' />)

    const separator = getByTestId('separator-test-id')

    expect(separator).toHaveStyle({height: 2, width: 2})
  })

  it('seperator verilen opacity ile render olmali', () => {
    const {separatorTestId} = testId
    const opacityValue = 0.5
    const {getByTestId} = render(
      <Separator
        testID={separatorTestId}
        size='full'
        orientation='vertical'
        opacity={opacityValue}
      />
    )
    const separatorElement = getByTestId(separatorTestId)

    expect(separatorElement).toHaveStyle({opacity: opacityValue})
  })
})

commonUiTestExtension(<Separator testID={testId.separatorTestId} />, testId.separatorTestId)
