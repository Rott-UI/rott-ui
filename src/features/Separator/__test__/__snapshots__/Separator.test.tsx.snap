// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator -> Custom Component separator component render olmalı ve snapshot eşleşmeli 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, 0)",
      "bottom": undefined,
      "flexGrow": undefined,
      "gap": undefined,
      "height": 2,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": 2,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "opacity": 1,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": 2,
      "zIndex": undefined,
    }
  }
/>
`;
